andela/travel_tool_front
# #160370625 Edit posted comments

#### What does this PR do?
This PR implements the edit comment functionality which enables users to edit own comment if errors were made in such comment.

#### Description of Task to be completed?
- Write tests for editing component and fix failing tests
- Create and implement the editing functionality for editing comments


#### How should this be manually tested?
- Clone the repository
- Run `make start` docker build command on the terminal to install all dependencies and run the App in the docker development environment server.
- On the browser, type the localhost URL ` http://localhost:3000` and Login with your Andela email account into the App.
- On the Requests page, click on the request Id from the list of requests displayed on the request table, you should be able to see the Request details Modal displayed.
- Add a new comment using the comment box.
- On the displayed added comment, click on the `edit` button to edit the comment.
- When done with editing and saving the changes, you should be able to see the edited comment displayed and labeled `edited`.


#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#160370625](https://www.pivotaltracker.com/story/show/160370625)

#### Screenshots (if appropriate)

Screenshot showing newly added comments

<img width="1276" alt="screen shot 2018-09-18 at 23 32 40" src="https://user-images.githubusercontent.com/12033494/45714873-87f36380-bb9b-11e8-9507-3a09c39f6327.png">









Screenshot showing a comment being edited

<img width="1276" alt="screen shot 2018-09-18 at 23 33 20" src="https://user-images.githubusercontent.com/12033494/45714906-9b063380-bb9b-11e8-9c1c-0266c6c5b13d.png">










Screenshot showing edited comment

<img width="1277" alt="screen shot 2018-09-18 at 23 34 41" src="https://user-images.githubusercontent.com/12033494/45714919-9fcae780-bb9b-11e8-94aa-c988c9f9ba9d.png">

#### Questions:
None
